plugins/modules/proxysql_replication_hostgroups.py

```diff
@@ -248,23 +248,38 @@ def create_repl_group(self, check_mode, result, cursor):
                              " mysql_replication_hostgroups, however" +
                              " check_mode is enabled.")
 
-    def update_repl_group(self, check_mode, result, cursor):
-        if not check_mode:
-            result['changed'] = \
-                self.update_repl_group_config(cursor)
-            result['msg'] = "Updated server in mysql_hosts"
-            result['repl_group'] = \
-                self.get_repl_group_config(cursor)
-            self.manage_config(cursor,
-                               result['changed'])
-        else:
+    def update_repl_group(self, result, cursor):
+        current = self.get_repl_group_config(cursor)
+
+        if self.check_type_support:
+            if current.get('check_type') != self.check_type:
+                result['changed'] = True
+                result['msg'] = "Updated replication hostgroups in check_mode"
+                if not self.check_mode:
+                    result['msg'] = "Updated replication hostgroups"
+                    self.update_check_type(cursor)
```

Comment:
```suggestion
                if not self.check_mode:
                    result['msg'] = "Updated replication hostgroups"
                    self.update_check_type(cursor)
                else:
                    result['msg'] = "Updated replication hostgroups in check_mode"
```
I can see that the result will be the same anyway. Just a suggestion:) A question of personal preferences